src/main/scala/de/frosner/ddq/constraints/JoinableConstraint.scala

```diff
@@ -2,37 +2,72 @@ package de.frosner.ddq.constraints
 
 import org.apache.spark.sql.{Column, DataFrame}
 
+import scala.util.Try
+
 case class JoinableConstraint(columnNames: Seq[(String, String)], referenceTable: DataFrame) extends Constraint {
+
   val fun = (df: DataFrame) => {
     val columnsMap = columnNames.toMap
     val renamedColumns = columnNames.map{ case (baseColumn, refColumn) => ("b_" + baseColumn, "r_" + refColumn)}
     val (baseColumns, refColumns) = columnNames.unzip
     val (renamedBaseColumns, renamedRefColumns) = renamedColumns.unzip
 
-    val nonUniqueRows = referenceTable.groupBy(refColumns.map(new Column(_)):_*).count.filter(new Column("count") > 1).count
+    val maybeNonUniqueRows = Try(
+      referenceTable.groupBy(refColumns.map(new Column(_)):_*).count.filter(new Column("count") > 1).count
+    )
 
     // rename all columns to avoid ambiguous column references
-    val renamedDf = df.select(baseColumns.zip(renamedBaseColumns).map {
+    val maybeRenamedDf = maybeNonUniqueRows.map(_ => df.select(baseColumns.zip(renamedBaseColumns).map {
       case (original, renamed) => new Column(original).as(renamed)
-    }:_*)
-    val renamedRef = referenceTable.select(refColumns.zip(renamedRefColumns).map {
+    }:_*))
+    val maybeRenamedRef = maybeNonUniqueRows.map(_ => referenceTable.select(refColumns.zip(renamedRefColumns).map {
       case (original, renamed) => new Column(original).as(renamed)
-    }:_*)
+    }:_*))
 
     // check if join yields some values
-    val renamedDfDistinct = renamedDf.distinct
-    val distinctBefore = renamedDfDistinct.count
-    val join = renamedDfDistinct.join(renamedRef, renamedColumns.map{
+    val maybeRenamedDfDistinct = maybeRenamedDf.map(_.distinct)
+    val maybeDistinctBefore = maybeRenamedDfDistinct.map(_.count)
+    val maybeJoin = maybeRenamedDfDistinct.map(_.join(maybeRenamedRef.get, renamedColumns.map{
       case (baseColumn, refColumn) => new Column(baseColumn) === new Column(refColumn)
-    }.reduce(_ && _))
-    val matchingRows = join.distinct.count
-    val matchedKeysPercentage = ((matchingRows.toDouble / distinctBefore) * 100).round
+    }.reduce(_ && _)))
+    val maybeMatchingRows = maybeJoin.map(_.distinct.count)
+    val maybeMatchedKeysPercentage = maybeMatchingRows.map(matchingRows =>
+        ((matchingRows.toDouble / maybeDistinctBefore.get) * 100).round)
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Usage of get on optional type.](https://www.codacy.com/app/frank_7/drunken-data-quality/file/2305054517/issues/source?bid=3033737&fileBranchId=3196266#l35)